ql/src/java/org/apache/hadoop/hive/ql/exec/tez/DagUtils.java

```diff
@@ -1297,18 +1297,20 @@ public static FileStatus validateTargetDir(Path path, Configuration conf) throws
   }
 
   // the api that finds the jar being used by this class on disk
-  public String getExecJarPathLocal(Configuration configuration) throws URISyntaxException {
-    // returns the location on disc of the jar of this class.
-
-    URI uri = DagUtils.class.getProtectionDomain().getCodeSource().getLocation().toURI();
-    if (configuration.getBoolean(ConfVars.HIVE_IN_TEST_IDE.varname, false)) {
-      if (new File(uri.getPath()).isDirectory()) {
-        // IDE support for running tez jobs
-        uri = createEmptyArchive();
+  public String getExecJarPathLocal(Configuration configuration) {
+    try {
+      // returns the location on disc of the jar of this class.
+      String uri = DagUtils.class.getProtectionDomain().getCodeSource().getLocation().toURI().toString();
+      if (uri.endsWith(".jar")) {
```

Comment:
can we get NPE here?

Reply:
Should not, but modified the catch block to Exception class, just to make sure no exception is thrown from this method.